### PR TITLE
Fix skip logic preview and falsy value

### PR DESCRIPTION
**Pull request by wwqrd**
- Use toggle for boolean value, which auto sets to false on mount.
- Update field validation to accept boolean values, or 0

Resolves https://github.com/codaco/Architect/issues/508

### CODE REVIEW

**Comment by rebeccamadsen** [comment]
This fixes the ui for me, and the false value doesn't have to toggle to be false. However, it shows "required" when the value is false in the form:

false

I can still Save though, so I'm not sure what the complaint is there.

**Comment by wwqrd** [comment]
I updated the ‘required’ validation to resolve but removed when I saw some overlap with another validator. Will check back on this when I know

**Comment by wwqrd** [comment]
Okay, I've update the validation rule. Along with another bug I found when using the number input.

Should also resolve #506

File: src/utils/validations.js [hunk 1]
```diff
@@ -13,9 +13,17 @@ const coerceArray = (value) => {
...
+
+  return !isNil(value);
+};
+
 export const required = () =>
   value =>
-    (value ? undefined : 'Required');
+    (hasValue(value) ? undefined : 'Required');
 
 export const requiredAcceptsZero = () =>
```
Comment (jthrilly) [inline src/utils/validations.js]: Should this be removed, now?

**Pull request merged**